radixdlt/src/main/java/com/radixdlt/network/transport/tcp/TCPTransportControlImpl.java

```diff
@@ -57,38 +59,61 @@
 	private static final Logger log = LogManager.getLogger("transport.tcp");
 
 	@Sharable
-	private static class TCPConnectionHandlerChannelInbound extends ChannelInboundHandlerAdapter {
+	static class TCPConnectionHandlerChannelInbound extends ChannelInboundHandlerAdapter {
 		private final RateLimiter droppedChannelRateLimiter = RateLimiter.create(1.0);
 		private final AtomicLong droppedChannelCount = new AtomicLong();
+
 		private final Object lock = new Object();
 		private final AtomicInteger channelCount = new AtomicInteger();
 		private final Map<String, LinkedList<SocketChannel>> channelMap = Maps.newHashMap();
+		private final Map<String, CompletableFuture<TransportOutboundConnection>> pendingMap = Maps.newHashMap();
```

Comment:
Added support here for connections that are pending, but not completed.  This avoids multiple connections being made to the same host when several messages are sent quickly.

Typically seems to take around ~100ms to establish TCP connection from docker container to docker container.